### ISSUE TITLE
[Google Search API] Contextual Autocomplete API

### ISSUE THREAD

**Issue by aliayar** [serpapi/public-roadmap#1119]
Currently, our Autocomplete API doesn't return any suggested search terms for certain search keywords such as `iphone android照片互傳`2023 or `iphone android照片互傳app`.

However, after a contextually similar keyword has been searched, before-mentioned search keywords return auto suggested terms.

Example when the keywords are filled in search box on the home page:

![image](https://github.com/serpapi/public-roadmap/assets/64849049/9ed3c4cf-573a-4af7-990e-442be97706b8)

or Google:

![image](https://github.com/serpapi/public-roadmap/assets/64849049/c8b49afe-ed21-4048-8423-52c9ff31747e)

On the search results page where there are results for a similar keyword:

<img width="1286" alt="contextual" src="https://github.com/serpapi/public-roadmap/assets/64849049/38ee8b63-8b9c-43af-8ee5-8e39bcf0f0df">

I suspected that this might be related to `gl` and `hl` parameters and search term should return results when they are set correctly but that's not the case.

Those two keywords return auto complete results only on a search results page where a similar keyword was searched before.

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/102910000051207?view=List) | [Playground](https://serpapi.com/playground?engine=google_autocomplete&q=iphone+android%E7%85%A7%E7%89%87%E4%BA%92%E5%82%B32023&gl=tw&hl=zh-cn&client=chrome) | [Inspect](https://serpapi.com/searches/651ac876d036a7e153d2b337/inspect) | [Google](https://www.google.com/search?q=android%E7%85%A7%E7%89%87%E4%BA%92%E5%82%B32023)

**Comment by zyc9012** [comment]
It's actually another form of "Related Searches". It's not fetched through the Autocomplete API but the Google search results.

![image](https://github.com/serpapi/public-roadmap/assets/3034310/49aa48a6-e80a-4028-8d77-dc78bad96e2f)

**Comment by aliayar** [comment]
It seems that `Autocomplete` part has more search terms than `related_searches`. What do you think?

**Comment by zyc9012** [comment]
> It seems that `Autocomplete` part has more search terms than `related_searches`. What do you think?

Yes. I'm wondering if we should parse this into another attribute, as it seems to be largely intersected with `related_searches`. wdty? @aciddjus @ilyazub @kagermanov27

**Comment by scxliu** [comment]
Hello , any update on this?

**Comment by tanys123** [comment]
Hi @scxliu, we have added a new client `gws-wiz` for awhile now. We have observed it produces similar results to Google SERP page. Can you try it out on your query? And let us know if they works for you.

**Comment by NateSkiles** [comment]
@tanys123 @zyc9012 wanted to tag you here as a customer asked about this today (autocomplete results returned on Google SERPs). Based on @zyc9012's screenshots, it looks like this previously matched the related queries, but I'm seeing largely different results between the two now:

<img width="1442" height="827" alt="Image" src="https://github.com/user-attachments/assets/25ff4c61-f98c-4d00-a60b-5d899e31cae2" />

> [..] we have added a new client gws-wiz for awhile now. We have observed it produces similar results to Google SERP page. 

I'm also seeing results vary significantly from the SERP when using the `gws-wiz` client parameter, but I'm assuming this is because there isn't a way to pass a location to the Autocomplete API.

Let me know if I'm missing something here, but if not, can we take another look at this? Thanks!

[Intercom](https://app.intercom.com/a/inbox/yzck75jr/inbox/shared/all/conversation/215473523952788?view=List)

**Comment by tanysheng** [comment]
Looks like it is a new client types in the autocomplete endpoint, we can potentially support it.

(default list before modifying the query)
<img width="1228" height="754" alt="Image" src="https://github.com/user-attachments/assets/d9c27410-7771-4ffc-b8a4-bc9697620457" />

(subsequent search)
<img width="1189" height="796" alt="Image" src="https://github.com/user-attachments/assets/702d6343-f808-4608-b256-4fbbf93bef3f" />
